rpcs3/rpcs3qt/game_list_frame.cpp

```diff
@@ -1194,54 +1272,87 @@ void game_list_frame::ShowContextMenu(const QPoint &pos)
 		});
 	}
 
-	bool has_hdd1_cache = false;
-	const std::string hdd1 = rpcs3::utils::get_hdd1_dir() + "/caches/";
-
-	for (const auto& entry : fs::dir(hdd1))
-	{
-		if (entry.is_directory && entry.name.starts_with(current_game.serial))
-		{
-			has_hdd1_cache = true;
-			break;
-		}
-	}
-
-	if (has_hdd1_cache)
+	const std::string hdd1_cache_base_dir = rpcs3::utils::get_hdd1_dir() + "/caches/";
+	bool has_hdd1_cache_dir = GetFirstSubDirBySerial(hdd1_cache_base_dir, current_game.serial) != "";
+	
+	if (has_hdd1_cache_dir)
 	{
 		QAction* remove_hdd1_cache = remove_menu->addAction(tr("&Remove HDD1 Cache"));
 		remove_hdd1_cache->setEnabled(!is_current_running_game);
-		connect(remove_hdd1_cache, &QAction::triggered, [this, hdd1, serial = current_game.serial]()
+		connect(remove_hdd1_cache, &QAction::triggered, [this, hdd1_cache_base_dir, serial = current_game.serial]()
 		{
-			RemoveHDD1Cache(hdd1, serial, true);
+			RemoveHDD1Cache(hdd1_cache_base_dir, serial, true);
 		});
 	}
 
-	if (has_cache_dir || has_hdd1_cache)
+	if (has_cache_dir || has_hdd1_cache_dir)
 	{
 		QAction* remove_all_caches = remove_menu->addAction(tr("&Remove All Caches"));
 		remove_all_caches->setEnabled(!is_current_running_game);
-		connect(remove_all_caches, &QAction::triggered, [this, current_game, cache_base_dir, hdd1]()
+		connect(remove_all_caches, &QAction::triggered, [this, current_game, cache_base_dir, hdd1_cache_base_dir]()
 		{
 			if (is_game_running(current_game.serial))
 				return;
 
 			if (QMessageBox::question(this, tr("Confirm Removal"), tr("Remove all caches?")) != QMessageBox::Yes)
 				return;
 
-			if (fs::remove_all(cache_base_dir))
-				game_list_log.success("Removed cache directory: '%s'", cache_base_dir);
-			else
-				game_list_log.error("Could not remove cache directory: '%s' (%s)", cache_base_dir, fs::g_tls_error);
-
-			RemoveHDD1Cache(hdd1, current_game.serial);
+			RemoveContentPath(cache_base_dir, "cache");
+			RemoveHDD1Cache(hdd1_cache_base_dir, current_game.serial);
 		});
 	}
 
+	// Disable the Remove menu if empty
+	if (!gameinfo->hasCustomConfig && !gameinfo->hasCustomPadConfig && !has_cache_dir && !has_hdd1_cache_dir)
+	{
+		remove_menu->setEnabled(false);
+	}
```

Comment:
I think you can just do this:
```suggestion
	remove_menu->setEnabled(!remove_menu->isEmpty());
```